src/bluefinClient.ts

```diff
@@ -2303,9 +2303,61 @@ export class BluefinClient {
     vaultName: string,
     amount?: number
   ): Promise<ResponseSchema> => {
-    if (amount) {
-      return this.interactorCalls.depositToVaultContractCall(amount, vaultName);
+    const coinHavingBalance = (
+      await this.contractCalls.onChainCalls.getUSDCoinHavingBalance(
+        {
+          amount,
+          address: this.walletAddress,
+        },
+        this.signer
+      )
+    )?.coinObjectId;
+    if (coinHavingBalance) {
+      if (amount) {
```

Comment:
Why is the amount an optional argument to this method? If a user is trying to deposit then they will specify the exact amount they want to deposit.

Reply:
Done